src/include/storage/storage_defs.h

```diff
@@ -259,6 +269,51 @@ class VarlenEntry {
 // To make sure our explicit padding is not screwing up the layout
 static_assert(sizeof(VarlenEntry) == 16, "size of the class should be 16 bytes");
 
+/**
+ * Equality checker that checks the underlying varlen bytes are equal (deep)
+ */
+struct VarlenContentDeepEqual {
```

Comment:
I think `std::hash` and `std::equal_to` should only be used for "default" behavior, because they are default template arguments to the STL data structures and people typically don't even pay attention to them when using STL.

I would argue that deep comparison that ignores ownership information is not "default" behavior, and I would rather users be conscious about needing this behavior.